test/integration/integration-test.ts

```diff
@@ -71,7 +68,7 @@ function testTransaction(testcase, type, lastClosedLedgerVersion, prepared,
   });
 }
 
-function setup(server = 'wss://s1.ripple.com') {
+function setup(this: any, server = 'wss://s1.ripple.com') {
```

Comment:
I'm not familiar with this syntax? 

`this` isn't referring to anything. If we want iVars, should we refactor into a class? If we want a global var, then we should define that and drop the use of `this`.

Reply:
yup, this is a TypeScript-ism. More info here: https://github.com/microsoft/TypeScript/wiki/'this'-in-TypeScript#specify-type-of-this-in-function-signature

I'd like to go back and replace/refactor a lot of these "any" types, but in the interest of not letting this PR sit too long I just added type definitions for the code as it existed in this PR, and then will do a follow up for any refactoring.